src/main/java/core/stack/MateStack.java

```diff
@@ -1,19 +1,49 @@
+
 package core.stack;
 
+import java.util.Arrays;
+import java.util.EmptyStackException;
+
 public class MateStack<T> {
-    public void push(T value) {
+    private static final int DEFAULT_SIZE = 8;
+    private Object[] stack;
+    private int size = 0;
 
+    public MateStack() {
+        stack = new Object[DEFAULT_SIZE];
+        Arrays.fill(stack, null);
+    }
+
+    public void push(T value) {
+        if (size < stack.length) {
+            stack[size] = value;
+            size++;
+        } else {
+            throw new StackOverflowError();
+        }
     }
 
     public T peek() {
-        return null;
+        if (size > 0) {
+            return (T) stack[size - 1];
+        } else {
+            throw new EmptyStackException();
+        }
     }
 
     public T pop() {
-        return null;
+        if (size > 0) {
+            Object value = stack[size - 1];
```

Comment:
It's unnecessary to set the value to null before reducing the size. JVM garbage collector will clean up the unused memory automatically.